### PR TITLE
Add support for folders in Cloudinary

### LINKED ISSUE #2972: Add support for folders in Cloudinary
- [* ] **I have created my request on the Product Board before I submitted this issue**
- [ *] **I have looked at all the other requests on the Product Board before I submitted this issue**

**Please describe your feature request:**
Currently the `strapi-provider-upload-cloudinary` plugin does not, as far as I can tell, support defining a folder under which to upload your files. Currently all files uploaded via strapi just go into the root of the Cloudinary media library, which is far from ideal. This means that if I have several apps uploading files to my Cloudinary account, there's no way for me to identify which files have been uploaded from which app, and thus no way to e.g. delete all of the files uploaded from a given app.

See `folder` under the Upload API Reference for more info: https://cloudinary.com/documentation/image_upload_api_reference

Please allow users to configure a folder to which the uploads from a given environment should go to. This should be a very trivial fix and frankly I'm a bit surprised it's not in there by default :)

Thanks!

### PR DESCRIPTION
**Pull request by abdonrd**
#### Description of what you did:

Add support for folders in Cloudinary

Fix #2972

#### My PR is a:

- [ ] 💥 Breaking change
- [ ] 🐛 Bug fix
- [x] 💅 Enhancement
- [x] 🚀 New feature

#### Main update on the:

- [ ] Admin
- [ ] Documentation
- [x] Framework
- [x] Plugin

#### Manual testing done on the following databases:

- [x] Not applicable
- [ ] MongoDB
- [ ] MySQL
- [ ] Postgres
- [ ] SQLite

### CODE REVIEW

File: packages/strapi-plugin-upload/services/Upload.js [hunk 1]
```diff
@@ -171,6 +171,10 @@ module.exports = {
           return file;
         });
 
+        config.providerOptions = {
+          ...(details.folder && { folder: details.folder }),
```
Comment (abdonrd) [inline packages/strapi-plugin-upload/services/Upload.js]: How can I add the `folder` option to the field?

Comment (abdonrd) [reply packages/strapi-plugin-upload/services/Upload.js]: I imagine it here:

Screenshot 2020-01-23 at 12 36 18

**Comment by ChristopherLMiller** [comment]
Will this break in any way for older uploads already added?  I've always edited the upload plugin manually to add a folder option while waiting on a long term solution for now.

**Comment by abdonrd** [comment]
> Will this break in any way for older uploads already added? I've always edited the upload plugin manually to add a folder option while waiting on a long term solution for now.

No, it shouldn't break anything. 😄

**Comment by alexandrebodin** [comment]
Hi @abdonrd. Adding the folder support is a really complexe subject when we start considering the different providers... Right now you could add the option inside the provider configuration quite easily but it would just upload all images to a specific folder. 

To configure the upload folder via the content type builder is another story :/ We would need to allow a provider to change the content-type-builder interface so we do not display that option for provider without folder support. I think you will be better off using lifecycles to move the images in the right folder instead for now.

**Comment by alexandrebodin** [comment]
@abdonrd You can open a discussion about that instead of a PR ?

**Comment by alexandrebodin** [comment]
Closing the PR until we discuss it more :)

**Comment by SalahAdDin** [comment]
@aleksandryackovlev When?

**Comment by SalahAdDin** [comment]
@aleksandryackovlev When? This is a very important feature.

**Pull request closed without merging**